examples/pytorch-mnist-membership-inference/demo.ipynb

```diff
@@ -186,6 +186,74 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
+    "This example also makes use of the `evaluation` custom task plugin package stored locally under the `task-plugins/securingai_custom/evaluation` directory.\n",
+    "To register these custom task plugins, we first need to package them up into an archive.\n",
+    "For convenience, the `Makefile` provides a rule for creating the custom task plugins archive file, just run `make custom-plugins`,"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "\n",
+    "# Create the workflows.tar.gz file\n",
+    "make custom-plugins"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now that the `evaluation` custom task plugin package is packaged into an archive file, next we register it by uploading the file to the REST API.\n",
+    "Note that we need to provide the name to use for custom task plugin package, this name must be unique under the custom task plugins namespace.\n",
+    "For a full list of the custom task plugins, use `restapi_client.restapi_client.list_custom_task_plugins()`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "response_custom_plugins = restapi_client.get_custom_task_plugin(name=\"evaluation\")\n",
+    "#restapi_client.delete_custom_task_plugin(name=\"evaluation\")\n",
+    "if response_custom_plugins is None or \"Not Found\" in response_custom_plugins.get(\"message\", []):\n",
+    "    response_custom_plugins = restapi_client.upload_custom_plugin_package(\n",
+    "        custom_plugin_name=\"evaluation\",\n",
+    "        custom_plugin_file=CUSTOM_PLUGINS_EVALUATION_TAR_GZ,\n",
+    "    )\n",
+    "\n",
+    "response_custom_plugins"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If at any point you need to update one or more files within the `evaluation` plugin package, you will need to unregister/delete the custom task plugin first using the REST API.\n",
+    "This can be done as follows,"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\n",
+    "# Delete the 'evaluation' custom task plugin package\n",
+    "restapi_client.delete_custom_task_plugin(name=\"evaluation\")\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "After you have deleted the task plugin in the testbed, re-run the `make custom-plugins` code block to update the package archive, then upload the updated plugin by re-running the `restapi_client.upload_custom_plugin_packge` block.\n",
+    "\n",
```

Comment:
It looks like the contents of the "evaluation" custom task plugin in this example is different [from the one in the tensorflow-mnist-classifier example](https://github.com/usnistgov/dioptra/tree/29ee50438a001715d1ca0f3038527fe286e74719/examples/tensorflow-mnist-classifier/task-plugins/securingai_custom/evaluation). This will lead to a naming conflict if a user runs that notebook and this notebook on the same deployment.

How to fix:

1. Rename `task-plugins/securingai_custom/evaluation` to something else, such as `task-plugins/securingai_custom/pytorch_mi`.
2. In `demo.ipynb`, update the value of the `custom_plugin_name` argument passed to `restapi_client.upload_custom_plugin_package()` to match, as well as the explanatory text surrounding these code cells to replace references to `evaluation` accordingly.
3. Update your scripts in the `src` directory to use the new custom task plugin name, for example https://github.com/usnistgov/dioptra/blob/b38422bc5254df96ba09c8b156a1ac5edf3e3a9a/examples/pytorch-mnist-membership-inference/src/train_mi.py#L263 would potentially change to `f"{_CUSTOM_PLUGINS_IMPORT_PATH}.pytorch_mi"` (it should match the `custom_plugin_name` chosen in step 2).
4. Update the Makefile to match the folder name change. For example, https://github.com/usnistgov/dioptra/blob/b38422bc5254df96ba09c8b156a1ac5edf3e3a9a/examples/pytorch-mnist-membership-inference/Makefile#L77 would update to
   ```
   CUSTOM_PLUGIN_PYTORCH_MI_SRC_FILES := $(wildcard $(PLUGINS_CUSTOM_DIR)/pytorch_mi/*.py)
   ```
   More or less a search and replace of `evalution` → `pytorch_mi` and `EVALUATION_` → `PYTORCH_MI_`.

I believe those are the main things that would need to be updated to handle the folder name change.